lib/puppet/util/package/version/pip.rb

```diff
@@ -150,14 +150,16 @@ def compare(this, other)
         end
       elsif (this.is_a? Array) && !(other.is_a? Array)
         raise Puppet::Error, 'Cannot compare #{this} (Array) with #{other} (#{other.class}). Only ±Float::INFINITY accepted.' unless other.abs == Float::INFINITY
-        return this.first.to_f <=> other
+        comparison = this.first.to_f <=> other
+        return comparison unless comparison == 0
+        return 1
       elsif !(this.is_a? Array) && (other.is_a? Array)
         raise Puppet::Error, 'Cannot compare #{this} (#{this.class}) with #{other} (Array). Only ±Float::INFINITY accepted.' unless this.abs == Float::INFINITY
-        return this <=> other.first.to_f
-      else
-        return this <=> other
+        comparison = this <=> other.first.to_f
+        return comparison unless comparison == 0
+        return -1
       end
-      0
+      this <=> other
```

Comment:
are this changes covered by the new versions added on pip_spec.rb?

Reply:
The new versions added in the `versions` array of `pip_spec.rb` uncovered the need for the changes done in this commit. Those and the other versions should cover the comparison functionality.